jupiterone/resource_framework.go

```diff
@@ -0,0 +1,297 @@
+	resp.TypeName = req.ProviderTypeName + "_framework"
+}
+
+// Configure implements resource.ResourceWithConfigure
+func (r *ComplianceFrameworkResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	p, ok := req.ProviderData.(*JupiterOneProvider)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected JupiterOneProvider, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.version = p.version
+	r.qlient = p.Qlient
+}
+
+// Schema implements resource.Resource
+func (*ComplianceFrameworkResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "A custom compliance standard, benchmark, or questionnaire.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The framework's name",
+			},
+			"version": schema.StringAttribute{
+				Required:    true,
+				Description: "Version of the framework itself (not an J1 API version)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"framework_type": schema.StringAttribute{
+				Required:    true,
+				Description: "Whether this is a standard, benchmark, or questionnaire",
+				Validators: []validator.String{
+					stringvalidator.OneOf(FrameworkTypes...),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"web_link": schema.StringAttribute{
+				Optional:    true,
+				Description: "A URL for referencing additional information about the framework",
+				// TODO: basic URL pattern matching
+				//Validators: []validator.String{
+				//},
+			},
+			"scope_filters": schema.ListAttribute{
+				Description: "JSON encoded filters for scoping the framework.",
+				Optional:    true,
+				ElementType: types.StringType,
+				Validators: []validator.List{
+					jsonValidator{},
+				},
+				PlanModifiers: []planmodifier.List{
+					jsonIgnoreDiffPlanModifierList(),
+				},
+			},
+		},
+	}
+}
+
+// ModifyPlan implements resource.ResourceWithModifyPlan
+func (*ComplianceFrameworkResource) ModifyPlan(ctx context.Context, _ resource.ModifyPlanRequest, _ *resource.ModifyPlanResponse) {
+	tflog.Info(ctx, "TODO")
+}
+
+// ConfigValidators implements resource.ResourceWithConfigValidators
+func (*ComplianceFrameworkResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	tflog.Info(ctx, "TODO")
+	return nil
+}
```

Comment:
Should I make issues for these as well?

Reply:
I just removed these. I added them thinking that `version` was bumped by the server on changes like `version` on rules, but it turns out it's just a user configured field on these.